### PR TITLE
Updating Readme.md file

### LINKED ISSUE #50: Add meaningful content in the readme
- explain what this extension does
- links to some granite code doc
- add some gif
- how to test CI builds
- link to contributing guide

### PR DESCRIPTION
**Pull request by athirakm94**
updating Readme.md file
 
 Fixes #50

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,124 @@
-# vscode-granite
+<img src="/media/title.png" alt="ollamacontine" width="1000"/>
+
+<!-- ABOUT THE PROJECT -->
+## About The Project
+In this project,we are building local AI co-pilot using **Granite Code, Ollama, and Continue**,where we utilize a collection of open-source components to implement a feature-rich developer co-pilot in Visual Studio Code, all while ensuring data privacy and compliance with licensing requirements. We are integrating Ollama with Continue to seamlessly work with Visual Studio Code. This integration will assist developers by accelerating the development process, enhancing performance, and improving code quality efficiently.
```
Comment (fbricon) [inline README.md:5]: - add hyperlinks to Granite Code, Ollama and Continue websites
- add spaces after commas

File: README.md [hunk 2]
```diff
@@ -1,7 +1,124 @@
...
+- **Generated Material Licensing:** Many models, even those with permissive usage licenses, do not disclose their training data and therefore may produce output that is derived from training material with licensing restrictions.
+- **Cost:** Many of these tools are paid solutions that require investment by the organization. For larger organizations, this would often include paid support and maintenance contracts, which can be extremely costly and slow to negotiate.
+
+### Models and Workflow of Vscode-Granite
+
+vscode-granite uses the Granite Code model, which is optimized for enterprise software development workflows and performs well across a range of coding tasks (e.g., code generation, fixing, and explanation), making it a versatile "all-around" code model.
+
+Granite Code comes in a wide range of sizes to fit your workstation's available resources. Generally, the bigger the model, the better the results.
+
+**Recommendation:** Model Size 20B for chat, 8B for tab code completion.
```
Comment (fbricon) [inline README.md:29]: @jamescho72 are we sticking with those recs?

File: README.md [hunk 3]
```diff
@@ -1,7 +1,124 @@
...
+
+ ### Workflow of Vscode-Granite
+
+ <p align="center">
+<img src="/media/workflow.png" alt="workflow" width="400"/>
+ </p>
+ 
+### Installation Prerequisites:
+
+  * OS: MacOS
```
Comment (fbricon) [inline README.md:41]: Nope, extension works cross platform

File: README.md [hunk 4]
```diff
@@ -1,7 +1,124 @@
...
+<img src="/media/workflow.png" alt="workflow" width="400"/>
+ </p>
+ 
+### Installation Prerequisites:
+
+  * OS: MacOS 
+  * DISK SPACE :Minimum 30 GB 
+  * latest [Visual Studio Code](https://code.visualstudio.com/)
+  * [Node.js](https://nodejs.org/) v20.0.0 or higher
+  * optional: [gh](https://cli.github.com/) to manage pull requests.
```
Comment (fbricon) [inline README.md:45]: No this is for development only. Those belong to the contributing guide.

File: README.md [hunk 5]
```diff
@@ -1,7 +1,124 @@
...
+  * Install [continu.dev](https://marketplace.visualstudio.com/items?itemName=Continue.continue) extention on Visual studio code
```
Comment (fbricon) [inline README.md:46]:
```
The [Continue.dev](...) extension should be installed automatically as a dependency. If not, please install it manually.
```

File: README.md [hunk 6]
```diff
@@ -1,7 +1,124 @@
...
+        <li>Clone the repository:</li>
+        <pre><code>git clone https://github.com/redhat-developer/vscode-granite.git </code></pre>
+        <li>Navigate to the project directory:</li>
+        <pre><code>cd vscode-granite</code></pre>
+        <li>Install the NPM dependencies:</li>
+        <pre><code>npm run install:ci</code></pre>
+    </ol>
+</body>
+
+This will install the dependencies for the extension and its [webview(s)](https://code.visualstudio.com/api/extension-guides/webview)
```
Comment (fbricon) [inline README.md:64]: No, this belongs to the contributing guide. You should have a paragraph that links to it

File: README.md [hunk 7]
```diff
@@ -1,7 +1,124 @@
...
+ 
+### Installation Prerequisites:
+
+  * OS: MacOS 
+  * DISK SPACE :Minimum 30 GB 
+  * latest [Visual Studio Code](https://code.visualstudio.com/)
+  * [Node.js](https://nodejs.org/) v20.0.0 or higher
+  * optional: [gh](https://cli.github.com/) to manage pull requests.
+  * Install [continu.dev](https://marketplace.visualstudio.com/items?itemName=Continue.continue) extention on Visual studio code
+
```
Comment (fbricon) [inline README.md:47]: Until the extension is available in the Marketplaces, you can explain how you can download the vsix from the GH releases page, and how to install it.

File: README.md [hunk 8]
```diff
@@ -1,7 +1,124 @@
...
+        <li>Navigate to the project directory:</li>
+        <pre><code>cd vscode-granite</code></pre>
+        <li>Install the NPM dependencies:</li>
+        <pre><code>npm run install:ci</code></pre>
+    </ol>
+</body>
+
+This will install the dependencies for the extension and its [webview(s)](https://code.visualstudio.com/api/extension-guides/webview)
+
+#### Use Vscode-granite UI to install extention and models
```
Comment (fbricon) [inline README.md:66]: Setup the Granite Code models as your code assistant.

File: README.md [hunk 9]
```diff
@@ -1,7 +1,124 @@
...
+
+This project offers an intuitive and streamlined UI designed to simplify the installation and management of extensions and Granite models. This user-friendly interface enables developers to quickly set up and configure their environment without needing to dive into complex configurations or terminal commands
+
+ step 1: **Run the Extenstion**
+
+Open VSCode and navigate to the Debugging tab on the left sidebar. Click on "Run and Debug," then select "Run Extension" to launch the extension.
+
+![runextension](/media/installandrun.gif)
```
Comment (fbricon) [inline README.md:74]: No, that belongs to the contributing guide

File: README.md [hunk 10]
```diff
@@ -1,7 +1,124 @@
...
+        <pre><code>npm run install:ci</code></pre>
+    </ol>
+</body>
+
+This will install the dependencies for the extension and its [webview(s)](https://code.visualstudio.com/api/extension-guides/webview)
+
+#### Use Vscode-granite UI to install extention and models
+
+This project offers an intuitive and streamlined UI designed to simplify the installation and management of extensions and Granite models. This user-friendly interface enables developers to quickly set up and configure their environment without needing to dive into complex configurations or terminal commands
+
```
Comment (fbricon) [inline README.md:69]: The first time the vscode-granite extension starts, the Granite Code setup wizard will open. You can open this wizard any time, from the command palette, execute the "Granite: Setup Granite Code as code assistant" command.

File: README.md [hunk 11]
```diff
@@ -1,7 +1,124 @@
...
+Open VSCode and navigate to the Debugging tab on the left sidebar. Click on "Run and Debug," then select "Run Extension" to launch the extension.
+
+![runextension](/media/installandrun.gif)
+ 
+ step 2:  **Install ollama**
+
+Once the extension is running, a new window will prompt you to install Ollama. Simply click the "Install with home brew"(for mac users) button to initiate the installation process.
+
+![installollama](media/installollama.gif)
```
Comment (fbricon) [inline README.md:80]: If Ollama is not installed, installation options are displayed:

- Install with Homebrew: if Homebrew is detected on your machine (Mac/Linux)
- Install with Script: on Linux
- Install Manually: on all platforms

Installing via Homebrew or Script will run the installation command in a VS Code Terminal. If you install Ollama manually, you'll be redirected to the Ollama download page.

Once OIlama is installed, the page will be refreshed automatically.

File: README.md [hunk 12]
```diff
@@ -1,7 +1,124 @@
...
+
+* Easily understand code sections
+
+* Tab to autocomplete code suggestions
+
+* Refactor functions where you are coding
+
+* Ask questions about your codebase
+
+* Quickly use documentation as context
```
Comment (fbricon) [inline README.md:107]: move under "why ollama" section above

File: README.md [hunk 13]
```diff
@@ -1,7 +1,124 @@
...
+
+Once the extension is running, a new window will prompt you to install Ollama. Simply click the "Install with home brew"(for mac users) button to initiate the installation process.
+
+![installollama](media/installollama.gif)
+
+step 3: **Install granite models** 
+
+ select the Granite model you wish to install. Follow the on-screen instruction to complete the setup of your models.
+
+![installmodels](media/installmodels.gif)
```
Comment (fbricon) [inline README.md:86]: Once the models are pulled into Ollama, Continue will be configured automatically to use them, the Continue chat view will open.

**Comment by fbricon** [comment]
- please rebase your PR against main
- the UI [changed](https://github.com/redhat-developer/vscode-granite/pull/66), so please use newer screenshots

File: README.md (hunk 2 above)
Comment (fbricon) [reply README.md:29]: @athirakm94 so 8b/8b it is

File: README.md [hunk 14]
```diff
@@ -1,11 +1,110 @@
-# vscode-granite
+<img src="/media/title.png" alt="ollamacontine" width="1000"/>
+
+<!-- ABOUT THE PROJECT -->
+## About The Project
+In this project,we are building local AI co-pilot using [Granite Code](https://github.com/ibm-granite/granite-code-models), [Ollama](https://ollama.com/), and [Continue](https://www.continue.dev/), where we utilize a collection of open-source components to implement a feature-rich developer co-pilot in Visual Studio Code, all while ensuring data privacy and compliance with licensing requirements. We are integrating Ollama with Continue to seamlessly work with Visual Studio Code. This integration will assist developers by accelerating the development process, enhancing performance, and improving code quality efficiently.
```
Comment (fbricon) [inline README.md:5]: > We aim to implement a feature-rich local AI co-pilot for Visual Studio Code, by utilizing a collection of open-source components: [Granite Code](https://github.com/ibm-granite/granite-code-models), [Ollama](https://ollama.com/), and [Continue](https://www.continue.dev/). By running Granite models locally, we ensure compliance with data privacy and licensing requirements. Our integration focuses on working with Continue to seamlessly interact with Visual Studio Code, helping developers streamline their workflow, boost productivity, and write higher-quality code.

File: README.md [hunk 15]
```diff
@@ -1,11 +1,110 @@
...
+
+<p align="center">
+  <img src="/media/ollamacontinue.png" alt="Continue" style="vertical-align: middle; width: 800px;" />
+</p>
```
Comment (fbricon) [inline README.md:9]: please remove that image

File: README.md [hunk 16]
```diff
@@ -1,11 +1,110 @@
...
+
+* Easily understand code sections
+
+* Tab to autocomplete code suggestions
+
+* Refactor functions where you are coding
+
+* Ask questions about your codebase
+
+* Quickly use documentation as context
```
Comment (fbricon) [inline README.md:41]: remove empty lines

File: README.md [hunk 17]
```diff
@@ -1,11 +1,110 @@
...
+
+**Recommendation:** Model Size 8B for chat, 8B for tab code completion.
+
+For more details, refer to [Granite Code Models](https://github.com/ibm-granite/granite-code-models)
+
+ #### Workflow of Vscode-Granite
+
+ <p align="center">
+<img src="/media/workflow.png" alt="workflow" width="400"/>
+ </p>
```
Comment (fbricon) [inline README.md:59]: remove unnecessary paragraph

**Review by fbricon — CHANGES_REQUESTED** [review]
Please delete the unnecessary images.

File: README.md [hunk 18]
```diff
@@ -1,11 +1,110 @@
...
+
+<h4 align="left">Why Continue.dev</h3>
+
+<div>
+
+[Continue](https://docs.continue.dev) is the leading open-source AI code assistant. You can connect any models and any context to build custom autocomplete and chat experiences inside [VS Code](https://marketplace.visualstudio.com/items?itemName=Continue.continue) and [JetBrains](https://plugins.jetbrains.com/plugin/22707-continue-extension)
+
+</div>
+
+<p></p>
```
Comment (fbricon) [inline README.md:31]: don't use html tags. This is markdown

**Comment by athirakm94** [comment]
Hi @fbricon,  as  i have raised PR from main branch , i am planning to reset all the changes done in the main branch of my forked repository ,so there might be a chance to affect this PR as well , i will raise new PR from my feature branch after all the changes

**Pull request closed without merging**